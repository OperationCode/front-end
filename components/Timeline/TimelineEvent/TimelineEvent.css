.eventContainer {
  margin: 0 0 30px 0;
  border: 1px solid #c5c5c5;
  padding: 20px;
  border-radius: 5px;
}

.eventTitle {
  text-transform: capitalize;
}

@media screen and (max-width: 680px) {
  .eventTitle {
    font-size: 1.5rem;
  }
}
