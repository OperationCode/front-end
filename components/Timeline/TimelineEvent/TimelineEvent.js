import React from 'react';
import PropTypes from 'prop-types';
import styles from './TimelineEvent.css';

TimelineEvent.propTypes = {
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.node]).isRequired,
  title: PropTypes.string.isRequired,
};

function TimelineEvent({ content, title }) {
  return (
    <div className={styles.eventContainer}>
      <h4 className={styles.eventTitle}>{title}</h4>

      <div className={styles.eventContent}>{content}</div>
    </div>
  );
}

export default TimelineEvent;
