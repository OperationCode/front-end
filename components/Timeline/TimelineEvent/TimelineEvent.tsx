import styles from './TimelineEvent.module.css';

export type TimelineEventPropsType = {
  content: React.ReactNode | React.ReactNode[];
  title: string;
};

function TimelineEvent({ content, title }: TimelineEventPropsType) {
  return (
    <div className="mb-8 border-1 border-solid border-[#c5c5c5] p-5 rounded-md">
      <h4 className="capitalize md:text-2xl">{title}</h4>

      <div className={styles.eventContent}>{content}</div>
    </div>
  );
}

export default TimelineEvent;
