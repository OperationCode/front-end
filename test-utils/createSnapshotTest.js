/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
