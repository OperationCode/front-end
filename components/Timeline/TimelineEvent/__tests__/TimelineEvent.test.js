import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import TimelineEvent from '../TimelineEvent';

describe('TimelineEvent', () => {
  test('should render properly with all required props', () => {
    createSnapshotTest(<TimelineEvent title="test title" content="here is some test content" />);
  });
});
