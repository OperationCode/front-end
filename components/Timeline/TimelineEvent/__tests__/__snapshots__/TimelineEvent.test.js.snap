// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineEvent should render properly with all required props 1`] = `
<div
  className="eventContainer"
>
  <h4
    className="eventTitle"
  >
    test title
  </h4>
  <div
    className="eventContent"
  >
    here is some test content
  </div>
</div>
`;
